lib/push-jenkins-update.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const githubClient = require('./github-client')
+
+const jenkinsToGhStatusMap = {
+  'SUCCESS': {
+    state: 'success',
+    message: 'all tests passed'
+  },
+  'FAILURE': {
+    state: 'failure',
+    message: 'build failure'
+  },
+  'ABORTED': {
+    state: 'error',
+    message: 'build aborted'
+  },
+  'UNSTABLE': {
+    state: 'error',
+    message: 'build unstable'
+  }
+}
+
+const rxDisplayName = /node\-test\-commit\-(\w+)/
+
+function pushJenkinsUpdate (options, build) {
+  const statusOpts = extendWithBuildData(options, build)
+  const createGhStatus = createGhStatusFn(statusOpts)
+  const isPending = build.result === null
+  const matchedGhStatus = jenkinsToGhStatusMap[build.result]
+
+  if (isPending) {
+    createGhStatus('pending', 'build in progress')
```

Comment:
Perhaps I'm overthinking it but we could even use variations for the avatar, for instance operating system/identifier or type of job? (test, lint, commit message validator, ...)